src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -206,18 +223,33 @@ utxoFromTx tx@(Tx _ outs) =
 prefilterBlock
     :: Block
     -> Wallet s
-    -> (UTxO, Set TxIn, s)
-prefilterBlock b (Wallet !utxo _ _ !s) =
+    -> (UTxO, Set TxIn, Set Tx, s)
```

Comment:
I've extended this one to also return our transactions (and avoid having to re-traverse the block in order to construct the metadata).